_posts/2024/2024-02-06-about-caching-in-eo.md

```diff
@@ -0,0 +1,241 @@
+(building JAR files). Its distinctive feature is the use of a single universal plugin with the same functionality 
+as five separate plugins for the standard lifecycle, but with significantly fewer dependencies. As a result,
+it provides a much faster startup, more optimal operation, and lower resource consumption.
+This leads to a significant increase in performance when compiling complex projects with a large number of modules.
+
+* The [maven-build-cache-extension](https://maven.apache.org/extensions/maven-build-cache-extension/)
+is used for large Maven projects that have a significant number of small `modules`.
+A `module` refers to a subproject within a larger project.
+Each `module` has its own `pom.xm` file, and there is an aggregator `pom.xml` that consolidates all the `modules`.
+This plugin takes a key for a `module`, it encapsulates the essential aspects of the `module`,
+including the source code and the configuration of the plugins used within it.
+`Modules` with the same key are current or unchanged and the cache can efficiently restore them.
+Conversely, the cache seamlessly delegates the build work to the standard Maven core,
+without interfering with the build execution logic.
+`maven-build-cache-extension` ensures that only the changed `modules` within the project will rebuild.
+
+
+### EO build cache
+
+The EO code uses the `Maven` for building projects.
+For this purpose, there is the `eo-maven-plugin` containing the essential goals for working with EO code.
+As previously mentioned, the build of projects in Maven follows a specific order of phases. 
+Below is a diagram illustrating the main phases and their corresponding goals for the EO:
+
+<p align="center">
+  <img src="/images/EO.svg">
+</p>
+
+In [Picture 3](/images/EO.svg) the goals of the `eo-maven-plugin` are highlighted in green.
+
+
+However, the actual work with EO code takes place in `AssembleMojo`.
+`AssembleMojo` is the goal consisting of other goals that work with the EO file, as shown in
+[Picture 4](/images/AssembleMojo.svg).
+
+
+<p align="center">
+  <img src="/images/AssembleMojo.svg">
+</p>
+
+Each goal within `AssembleMojo` is a distinct compilation step for EO code.
+These tasks happen one after the other, and each task relies on the output of the one before it.
+Each task has directories for input and output data, as well as a directory for storing cached data.
+Using the program name, each task can receive and store data.
+
+
+The previous caching mechanism in EO made use of distinct interfaces, specifically `Footprint` and `Optimization`.
+These caching interfaces shared similar logic, but with minor differences.
+For instance, `Footprint` verifies the EO version of the compiler, whereas the remaining checks are identical.
+Additionally, the conditions for searching data in the cache had errors. 
+The cached file is considered valid if the end time of goal's execution
```

Comment:
@Yanich96 You don't need to explain it again. I've already mentioned this bug above. Could you please remove this sentence?